lib/maths/CTimeSeriesModel.cc

```diff
@@ -1619,6 +1619,8 @@ void CUnivariateTimeSeriesModel::reinitializeStateGivenNewComponent(
     // re-weight so that the total sample weight corresponds to the sample
     // weight the model receives from a fixed (shortish) time interval.
 
+    m_ResidualModel->removeModels(
```

Comment:
Perhaps add a comment to explain why we're removing Poisson from model selection in this case?

Reply:
I added this comment